graal/backends/core/analyzers/cloc.py

```diff
@@ -55,22 +52,42 @@ def analyze(self, **kwargs):
         finally:
             subprocess._cleanup()
 
-        for line in msg.split("\n"):
+        if repository_level:
+            results = {}
+        else:
+            results = {
+                "blanks": 0,
+                "comments": 0,
+                "loc": 0
+            }
+
+        for line in msg.strip().split("\n"):
```

Comment:
> Instead of adding more logic to the analyze method, I would create two private methods, one works for the single file and one for the full repo, this would probably make the code simpler to read.

Agreed. Thanks for the suggestion :)

Reply:
that sound perfect, maybe `analyze_file()` and `analyze_repository()` should be private methods (e.g., `__analyze_file()`)